lib/frame.rb

```diff
@@ -0,0 +1,32 @@
+class Frame
+
+  def initialize(frame)
+    fail "invalid frame" if (frame.length == 2) && (frame[1] == 10)
+    fail "invalid frame" if (frame.length == 1) && (frame[0] != 10)
+    @frame = frame
+  end
+
+  def score
+    @frame.sum
+  end
+
+  def to_a
+    @frame
+  end
```

Comment:
If you have a couple of these, you might want to use attr_accessor (or reader) to shorten this bit of code.